docs/inventories.md

```diff
@@ -0,0 +1,205 @@
+
+```
+[cluster_worker_nodes]
+host-1.example.com host_template=MasterWorker
+```
+
+## Optional Groups
+
+### CDSW
+
+Servers which will form a Cloudera Data Science Workbench (CDSW) cluster need to be added in the following way. The aggregate group `cdsw` is **important**. This ensures that the special pre-requisite tasks for CDSW are correctly applied.
+
+```ini
+[cdsw_master_nodes]
+cdsw-1.example.com host_template=CDSW-Master
+
+[cdsw_worker_nodes]
+cdsw-2.example.com
+cdsw-3.example.com
+cdsw-4.example.com
+
+[cdsw_worker_nodes:vars]
+host_template=CDSW-Workers
+
+[cdsw:children]
+cdsw_master_nodes
+cdsw_worker_nodes
+
+[cluster:children]
+cluster_master_nodes
+cluster_worker_nodes
+cdsw
+```
+
+### HDFS Encryption (KMS / Key Trustee Server)
+
+Configuring HDFS encryption requires two extra groups `kts_active` and `kms_servers` with a third, optional (but **recommended**) group `kts_passive` to enable Key Trustee Server high availability. 
+
+The `kts_active` and `kts_passive` groups must contain a single node each. The KMS group `kms_servers` must have at least one host but can have as many as desired. 
+
+```ini
+[kms_servers]
+kms-1.example.com
+kms-2.example.com
+kms-3.example.com
+
+[kts_active]
+kts-1.example.com
+
+[kts_passive]
+kts-2.example.com
+
+[cluster:children]
+cluster_master_nodes
+cluster_worker_nodes
+kts_active
+kts_passive
+kms_servers
+```
+
+### Infrastructure
+
+The following host groups are **optional**.
+
+If these are omitted or left empty, the steps to provision these items will be skipped. This is for cases where such infrastructure is already provided and we do not want to create it from scratch.
+
+#### Database server
+
+```ini
+[db_server]
+host-10.example.com
+```
+
+#### Kerberos KDC
+
+```ini
+[krb5_server]
+host-10.example.com
+```
+
+#### Load balancer (HAProxy)
+
+```ini
+[haproxy]
+host-10.example.com
+```
+
+#### TLS CA Server
+
+```ini
+[ca_server]
+host-10.example.com
+```
+
+#### Custom parcel repository
+
+```ini
+[custom_repo]
+host-10.example.com
+```
+
+## Multiple Clusters
+
+It is possible to define multiple clusters in the inventory. The key point is that the `cluster` group must contain **all** servers which will be under Cloudera Manager's control, regardless of which cluster they belong to. 
+
+
+The inventory group names `cluster1` and `cluster2` are arbitrary. This is just a convenience to make the inventory easier to understand. 
+
+```ini
+[cluster1]
+host-2.example.com  host_template=Cluster1-Master
+host-3.example.com  host_template=Cluster1-Worker
+host-4.example.com  host_template=Cluster1-Worker
+host-5.example.com  host_template=Cluster1-Worker
+
+[cluster2]
+host-6.example.com  host_template=Cluster2-Master
+host-7.example.com  host_template=Cluster2-Worker
+host-8.example.com  host_template=Cluster2-Worker
+host-9.example.com  host_template=Cluster2-Worker
+
+[cluster:children]
+cluster1
+cluster2
+```
```

Comment:
it would be good to add the [deployment:children] to encapsulate variables for the whole deployment as separate from localhost for when the user wants to apply things like tls=True